client/state/plugins/installed/selectors-ts.ts

```diff
@@ -95,3 +130,61 @@ export const getAllPluginsIndexedBySiteId = createSelector(
 	},
 	( state ) => [ getAllPluginsIndexedByPluginSlug( state ), getSiteIdsThatHavePlugins( state ) ]
 ) as { ( state: AppState ): { [ siteId: number ]: { [ pluginSlug: string ]: Plugin } } };
+
+export const getFilteredAndSortedPlugins = createSelector(
+	( state: AppState, siteIds: number[], pluginFilter?: PluginFilter ) => {
+		const allPluginsIndexedBySiteId = getAllPluginsIndexedBySiteId( state );
+
+		// Properties on the objects in allPluginsIndexedBySiteId will be modified and the
+		// selector memoization always returns the same object, so use cloneDeep to avoid
+		// altering it for everyone.
+		const allPluginsForSites: { [ pluginSlug: string ]: Plugin } = cloneDeep(
+			siteIds
+				.map( ( siteId: number ) => allPluginsIndexedBySiteId[ siteId ] )
+				.filter( Boolean )
+				.reduce( ( accumulator, current ) => ( { ...accumulator, ...current } ), {} )
+		);
+
+		// Filter the sites object on the plugins so that only data for the requested siteIds is present
```

Comment:
Can you elaborate on this one? I'm not sure I understand how this is related to data only for requested site IDs.

Reply:
In my opinion it shouldn't be, but that is how the code was written before. When I first refactored I left the site ID filtering out and it broke a lot of component code which was assuming that this selector filtered out the other sites for it. This was one of the major difficulties I had, so eventually I decided to just preserve the existing behavior and comment it to make it more obvious.

Reply:
It's worth spending some more time, perhaps in another PR, to make this clearer, even if it's just with some additional inline comments and/or additional unit tests.